app/src/main/java/com/example/instagramclone/fragments/main_screen/compose/PostItem.kt

```diff
@@ -0,0 +1,215 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.media3.exoplayer.ExoPlayer
+import coil.compose.AsyncImage
+import com.example.domain.entities.Post
+import com.example.instagramclone.R.drawable
+@androidx.annotation.OptIn(androidx.media3.common.util.UnstableApi::class)
+@Composable
+fun PostItem(
+    modifier: Modifier = Modifier,
+    postItem: Post,
+    savePost: () -> Unit,
+    exoPlayer: ExoPlayer,
+    isVisible: Boolean
+) {
+    Surface(
+        modifier = modifier.fillMaxSize(),
+        color = Color.White
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+        ) {
+            PostTopbar(postItem = postItem, modifier = Modifier)
+            PostItemContent(exoPlayer = exoPlayer, postItem = postItem, isVisible = isVisible)
+            PostBottomBar(savePost = { savePost() })
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = "2,426" + " likes",
+                fontWeight = FontWeight.Bold
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = buildAnnotatedString {
+                    withStyle(style = SpanStyle(fontWeight = FontWeight.Bold)) {
+                        append(postItem.userName)
+                        append(" ")
+                    }
+                    append(postItem.contentDesc)
+                }
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = "View all 83 comments",
+                style = TextStyle(color = Color.Gray)
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
```

Comment:
All the colors should be coming from `MaterialTheme.colorScheme` to make it easier for us to switch between light/dark themes.

And it's best to use the [4-point grid system](https://medium.com/@aratidube12lns/4-point-grid-system-for-more-consistent-interface-design-efea81dea3f3#:~:text=What%20is%20the%204%2Dpoint,padding%2C%20margins%20and%20line%20heights.) wherever possible.